### DISCUSSION TITLE
Shouldn't the `StatementCache` be tied to a single database session?

### DISCUSSION THREAD

**Discussion by k-anshul** [jeroenrinzema/psql-wire#68] [Q&A] [answered]
I see that the `StatementCache` is currently global in the code which means Statements from one connection can be shared with other connections. 
Shouldn't it be tied to a single connection as specified in the postgres docs : https://www.postgresql.org/docs/current/sql-prepare.html

**Comment by jeroenrinzema** [comment, marked as answer]
Hi @k-anshul, good catch! I will implement this change.

**Reply by jeroenrinzema** [reply]
@k-anshul the change has been implemented in v0.12.1
